pandas/core/frame.py

```diff
@@ -3472,10 +3490,13 @@ def query(self, expr, inplace=False, **kwargs):
 
         if inplace:
             self._update_inplace(result)
+            return None
         else:
             return result
 
-    def eval(self, expr, inplace=False, **kwargs):
+    def eval(
+        self, expr: str, inplace: bool = False, **kwargs
+    ) -> Optional[Union[AnyArrayLike, Scalar]]:
```

Comment:
the output type here is suspect

Reply:
Looking

Reply:
From docstring:
```
        Returns
        -------
        ndarray, scalar, pandas object, or None
            The result of the evaluation or None if ``inplace=True``.
```
so does `Optional[Union[AnyArrayLike, DataFrame, Scalar]]` work?